src/main/scala/BootAddrReg.scala

```diff
@@ -0,0 +1,38 @@
+package testchipip
+
+import chisel3._
+import chisel3.experimental.{IO}
+import freechips.rocketchip.config.{Parameters, Field}
+import freechips.rocketchip.diplomacy._
+import freechips.rocketchip.tilelink._
+import freechips.rocketchip.devices.tilelink._
+import freechips.rocketchip.regmapper._
+import freechips.rocketchip.subsystem._
+import freechips.rocketchip.util._
+import freechips.rocketchip.tile._
+import freechips.rocketchip.prci._
+
+
+case class BootAddrRegParams(
+  defaultBootAddress: BigInt = 0x80000000L, // This should be DRAM_BASE
+  bootRegAddress: BigInt = 0x4000,
+  slaveWhere: TLBusWrapperLocation = PBUS
+)
+case object BootAddrRegKey extends Field[BootAddrRegParams](BootAddrRegParams())
+
+trait HasPeripheryBootAddrReg { this: BaseSubsystem =>
+  val params = p(BootAddrRegKey)
+
+  val tlbus = locateTLBusWrapper(params.slaveWhere)
+
+  val device = new SimpleDevice("boot-address-reg", Nil)
+
+  tlbus {
+    val node = TLRegisterNode(Seq(AddressSet(params.bootRegAddress, 4096-1)), device, "reg/control", beatBytes=tlbus.beatBytes)
+    tlbus.toVariableWidthSlave(Some("boot-address-reg")) { node := TLBuffer() }
```

Comment:
Stupid question, do we need a buffer here? Can we just directly wire the register to the PBUS? I think this is simple enough to not need any buffering.

Reply:
Probably no buffer is necessary.